configure.ac

```
@@ -5346,12 +5342,23 @@ PKG_CHECK_MODULES([ZLIB], [zlib >= 1.2.0], [
     ])
   ])
 ])
-
-dnl binascii can use zlib for optimized crc32.
-AS_VAR_IF([have_zlib], [yes], [
-  BINASCII_CFLAGS="-DUSE_ZLIB_CRC32 $ZLIB_CFLAGS"
-  BINASCII_LIBS="$ZLIB_LIBS"
-])
+dnl temporarily throw only errors on builds other than wasi until wasi supports zlib
+AS_CASE([$ac_sys_system],
+  [WASI], [
+    BINASCII_CFLAGS="-DNO_ZLIB_CRC32"
+    BINASCII_LIBS=""
+  ],
+  [*], [
+    if test "$have_zlib" != "yes"; then
+      AC_MSG_ERROR([m4_normalize([
+        zlib.h and libz are required. Install your OS's zlib-devel or
+        zlib1g-dev equivalent library or get it from https://zlib.net/.
+      ])])
```

Let's not make it a hard failure yet.
```suggestion
      AC_MSG_WARN([m4_normalize([
        zlib (zlib.h and libz) was not found. Building CPython
        without zlib is not supported, although it might work
        for limited use cases.
        Install your OS's zlib-devel or zlib1g-dev equivalent
        library or get it from https://zlib.net/.
      ])])
```